png2asm.py

```diff
@@ -3,31 +3,44 @@
 from sys import exit, argv
 from os import path
 
+TOKEN_LIMIT = 48 # 50 Max Tokens - 2 Tokens for variable name and type
+
 def print_and_quit(message):
     print(message)
     input("Press anything to quit...")
     exit()
 
 def path_to_basename(file_path):
     return path.splitext(file_path)[0]
+    
+def write_variable_to_file(file, variableName, hexels, lowLimit, highLimit, height):
+    for y in range(height):
+        if y != 0:
+            file.write(" "*len(variableName) + " DD ")
+        for x in range(lowLimit, highLimit):
+            file.write(hexels[height*x + y])
+            file.write(", ")
+        file.write(hexels[height*x + y])
+        file.write("\n")
 
 def write_hexels_to_file(fileName, hexels, width, height):
     outputName = fileName + '.inc'
-    variableName = "var" + fileName
     f = open(outputName, "w")
-    if f:
-        f.write(variableName + " DD ")
-        for y in range(height):
-            if y != 0:
-                f.write(" "*len(variableName) + " DD ")
-            for x in range(width - 1):
-                f.write(hexels[height*x + y])
-                f.write(", ")
-            f.write(hexels[height*x + y])
+
+    if f: 
+        splitCount = height // TOKEN_LIMIT
+        print(splitCount)
+        for i in range(splitCount + 1):
+            variableName = "var" + fileName + "_" + str(i)
+            f.write(variableName + " DD ")
```

Comment:
shouldn't this part where writing in the file be contained in the write_variable method?

Reply:
Great catch. I updated it. Also removed the print(splitCount).